libbeat/processors/timestamp/timestamp_test.go

```diff
@@ -199,3 +199,87 @@ func TestBuiltInTest(t *testing.T) {
 		assert.Contains(t, err.Error(), "failed to parse test timestamp")
 	}
 }
+
+func TestTimezone(t *testing.T) {
+	cases := map[string]struct {
+		Timezone string
+		Expected time.Time
+	}{
+		"no timezone": {
+			Expected: expected,
+		},
+		"location label": {
+			// Use a location without DST to avoid surprises
+			Timezone: "America/Panama",
+			Expected: expected.Add(5 * time.Hour),
+		},
+		"UTC label": {
+			Timezone: "Etc/UTC",
+			Expected: expected,
+		},
+		"GMT label": {
+			Timezone: "Etc/GMT+2",
+			Expected: expected.Add(2 * time.Hour),
+		},
+		"UTC as standard offset": {
+			Timezone: "+0000",
+			Expected: expected,
+		},
+		"standard offset": {
+			Timezone: "+0430",
+			Expected: expected.Add(-4*time.Hour - 30*time.Minute),
+		},
+		"hour and minute offset": {
+			Timezone: "+03:00",
+			Expected: expected.Add(-3 * time.Hour),
+		},
+		"minute offset": {
+			Timezone: "+00:30",
+			Expected: expected.Add(-30 * time.Minute),
+		},
+		"abbreviated hour offset": {
+			Timezone: "+04",
+			Expected: expected.Add(-4 * time.Hour),
+		},
+		"negative hour and minute offset": {
+			Timezone: "-03:30",
+			Expected: expected.Add(3*time.Hour + 30*time.Minute),
+		},
+		"negative minute offset": {
+			Timezone: "-00:30",
+			Expected: expected.Add(30 * time.Minute),
+		},
+		"negative abbreviated hour offset": {
```

Comment:
maybe a test here with a given timezone format that's not supported? Like `Z0700` or `Z07:00`.